src/main/host/syscall/handler/time.rs

```diff
@@ -99,6 +99,40 @@ impl SyscallHandler {
         Ok(0.into())
     }
 
+    #[log_syscall(/* rv */ std::ffi::c_uint, /* seconds */ std::ffi::c_uint)]
+    pub fn alarm(ctx: &mut SyscallContext, seconds: std::ffi::c_uint) -> Result<u32, SyscallError> {
+        // We return the remaining time of any previously armed timer, or zero if there is none.
+        let prev_remaining = ctx
+            .objs
+            .process
+            .realtime_timer_borrow()
+            .remaining_time()
+            .unwrap_or(SimulationTime::ZERO);
+        let prev_remaining_secs = prev_remaining.as_secs();
```

Comment:
The kernel seems to round to the nearest second whereas this truncates the sub-second value. The kernel also seems to never return 0 if there is a timer (ex: 1 ns results in a return value of 1 sec).

```c
  /*
	 * We can't return 0 if we have an alarm pending ...  And we'd
	 * better return too much than too little anyway
	 */
	if ((!it_old.it_value.tv_sec && it_old.it_value.tv_nsec) ||
	      it_old.it_value.tv_nsec >= (NSEC_PER_SEC / 2))
		it_old.it_value.tv_sec++;

```

Reply:
Oof, good catch! Done